pkg/cmd/infra/router/template.go

```diff
@@ -648,11 +647,11 @@ func (o *TemplateRouterOptions) Run(stopCh <-chan struct{}) error {
 	var recorder controller.RejectionRecorder = controller.LogRejections
 	if o.UpdateStatus {
 		lease := writerlease.New(time.Minute, 3*time.Second)
-		go lease.Run(wait.NeverStop)
+		go lease.Run(stopCh)
```

Comment:
I think changing lines 650 and 654 to use the `stopCh` makes sense, but someone please correct me if i'm wrong.